cmd/simulator/load/loader.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/cmd/simulator/config"
+	"github.com/ava-labs/subnet-evm/cmd/simulator/key"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ethereum/go-ethereum/common"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// CreateLoader creates a WorkerGroup from [config] to perform the specified simulation.
+func CreateLoader(ctx context.Context, config config.Config) (*WorkerGroup, error) {
+	// Construct the arguments for the load simulator
+	if len(config.Endpoints) < config.Workers {
+		// Ensure there are at least [config.Workers] config.Endpoints by creating
+		// duplicates as needed.
+		for i := 0; len(config.Endpoints) < config.Workers; i++ {
+			config.Endpoints = append(config.Endpoints, config.Endpoints[i])
```

Comment:
could we avoid modifying the `config` object so it always corresponds to the command line arguments as provided?

Reply:
Updated